projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -817,19 +817,75 @@ private function render_file_field( $id, $label, $class, $required, $required_fi
 			$input_attributes .= $attr . ( $value !== '' ? '="' . $value . '"' : '' ) . ' ';
 		}
 
-		$field  = $this->render_label( 'file', $id, $label, $required, $required_field_text );
-		$field .= "<div class='jetpack-form-file-field__dropzone' data-id='" . esc_attr( $id ) . "'>\n";
-		$field .= "<a href='#' class='wp-block-button__link wp-element-button'>" . esc_html__( 'Select a file', 'jetpack-forms' ) . "</a>\n";
-		$field .= "<span class='jetpack-form-file-field__short'>" . esc_html__( '....or drag and drop a file.', 'jetpack-forms' ) . "</span>\n";
-		$field .= '<input ' . trim( $input_attributes ) . "/>\n";
-		$field .= "<input type='hidden' name='" . esc_attr( $id ) . "_token' class='jetpack-form-file-field__token' value='' />\n";
-		$field .= "<div class='jetpack-form-file-field__preview-wrap' aria-live='polite'></div>\n";
-		$field .= "</div>\n";
+		$max_file_size = wp_max_upload_size();
+		$file_size_units = array(
+			_x( 'B', 'unit symbol', 'jetpack-forms' ),
+			_x( 'KB', 'unit symbol', 'jetpack-forms' ),
+			_x( 'MB', 'unit symbol', 'jetpack-forms' ),
+			_x( 'GB', 'unit symbol', 'jetpack-forms' ),
+		);
+
+		$global_state = array(
+			'i18n'           => array(
```

Comment:
For i18n, if the values are not dynamic in the frontend, you can use [config](https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/api-reference/#wp_interactivity_config) instead of state.

```php
wp_interactivity_config( 'jpDropZone', $i18n );
```

```js
const { fileSizeUnits } = getConfig();
const sizes = fileSizeUnits || [ 'Bytes', 'KB', 'MB', 'GB', 'TB' ];
```

Reply:
@enejb updated this. Marking as resolved.